channels/smartcard/client/smartcard_main.c

```diff
@@ -258,7 +315,8 @@ static void smartcard_release_all_contexts(SMARTCARD_DEVICE* smartcard)
 static UINT smartcard_free(DEVICE* device)
 {
 	UINT error;
-	SMARTCARD_DEVICE* smartcard = (SMARTCARD_DEVICE*) device;
+	CHECKED_SMARTCARD_DEVICE(smartcard, device);
```

Comment:
Good idea, but maybe name it so it is obvious that this macro does a `return` in the error case?